src/main/java/com/fauna/client/FaunaClient.java

```diff
@@ -271,4 +280,22 @@ public <E> PageIterator<E> paginate(Query fql, Class<E> resultClass) {
     public <E> PageIterator<E> paginate(Query fql, Class<E> resultClass, QueryOptions options) {
         return new PageIterator<>(this, fql, resultClass, options);
     }
+
+    public <E> FaunaStream<E> stream(StreamRequest streamRequest, Class<E> resultClass) {
+        // TODO: I don't think there's a need for an asyncStream method, as the FaunaStream constructor returns
```

Comment:
I'm not so sure. The method below this makes a query and then blocks on on the tokenFuture before it can start the stream. Seems like a very good case for an asyncStream method, or you should handle it how you handled it here (i.e. make it non-blocking by passing futures until you're into the FaunaStream constructor)

Reply:
Yeah, I could see `asyncStream(Query fql...)` making sense, but I think this method doesn't need an `async` counterpart, because I don't think it blocks.